backend/controller/leases/leases.go

```diff
@@ -23,7 +25,14 @@ type Leaser interface {
 	//
 	// This function will return [ErrConflict] if a lease is already held. The [ttl]
 	// must be at _least_ 5 seconds.
-	AcquireLease(ctx context.Context, key Key, ttl time.Duration) (Lease, error)
+	//
+	// The returned context will be cancelled when the lease fails to renew.
+	AcquireLease(ctx context.Context, key Key, ttl time.Duration, metadata optional.Option[any]) (Lease, context.Context, error)
+
+	// GetLeaseInfo returns the metadata and expiry time for a key.
+	//
+	// metadata should be a pointer to the type that metadata should be unmarshaled into.
+	GetLeaseInfo(ctx context.Context, key Key, metadata any) (expiry time.Time, err error)
```

Comment:
Not particularly happy with the name for this `GetLeaseInfo()` func. Any better ideas?